### DISCUSSION TITLE
Adding a filetype to LazyVim

### DISCUSSION THREAD

**Discussion by duckpuppy** [LazyVim/LazyVim#3750] [Q&A]
I'm specifically trying to have Neovim see `*.tf` files as Terraform files even if they're empty. Currently - an empty `.tf` file will be the `tf` filetype. Once it has HCL code in it then NeoVim detects the filetype as `terraform`.

I tried putting this into my `options.lua`:

```lua
vim.filetype.add({
  extension = {
    tf = "terraform",
  },
})
```

It's still seeing an empty *.tf file as a `tf` filetype. Is there a better way to force it to always see a `.tf` file as Terraform?

**Comment by duckpuppy** [comment]
I'll add that OTHER lua code that I have in my `options.lua` is taking effect, just not that one snippet.

**Reply by dpetka2001** [reply]
Create a file `~/.config/nvim/filetype.lua` and put the contents of `vim.filetype.add` in there.

**Reply by matiasilva** [reply]
For any future readers, ^ works and is the recommended and most portable imo way of doing things.

**Comment by Riddlerrr** [comment]
Try to set it up [this way](https://gist.github.com/Riddlerrr/126784d605913c3c240fc21efdc57a41)

**Comment by norbertkeri** [comment]
For others finding this, another way of doing it:

```lua
-- .config/nvim/ftdetect/terraform.lua

if vim.filetype then
    vim.filetype.add({
        pattern = {
            ["*.tf"] = "terraform"
        }
    })
end
```
